modules/sc-mesh-secure-deployment/src/nats/src/comms_settings.py

```diff
@@ -0,0 +1,288 @@
+            self.logger.debug("load settings: %s, %s", ret, info)
+
+            #self.api_version = int(parameters["api_version"])
+            self.frequency, self.delay, self.device_amount= map(quote,
+                                                                (str(parameters["frequency"]),
+                                                                 str(parameters["delay"]),
+                                                                 str(parameters["amount"])))
+
+            if validation.validate_delay(self.delay) and validation.validate_frequency(int(self.frequency)):
+                ret, info  = "OK", "CSA settings OK"
+            else:
+                ret, info = "FAIL", "Invalid delay or frequency"
+
+            self.logger.debug(" settings validation: %s, %s", ret, info)
+            if ret == "FAIL":
+                self.comms_status.mesh_cfg_status = \
+                    comms.STATUS.mesh_cfg_not_stored
+                self.logger.error("save settings failed: %s, %s", ret, info)
+            else:
+                ret, info = self.__save_settings(path, file)
+                self.logger.debug("save settings: %s, %s", ret, info)
+                self.csa_state = 1
+                self.csa_count = 0
+
+                return ret, info, self.delay
+
+        except (json.decoder.JSONDecodeError, KeyError,
+                TypeError, AttributeError) as error:
+            self.comms_status.mesh_cfg_status = \
+                comms.STATUS.mesh_cfg_not_stored
+            ret, info = "FAIL", "JSON format not correct" + str(error)
+            self.logger.error("csa settings validation: %s, %s", ret, info)
+
+        return ret, info, self.delay
+
+    def handle_mesh_settings(self, msg: str, path="/opt",
+                             file="mesh_stored.conf") -> (str, str):
+        """
+        Handle mesh settings
+        """
+        try:
+            parameters = json.loads(msg)
+            print(parameters)
+            self.msversion = "nats"
+            self.api_version = int(parameters["api_version"])
+            self.ssid = quote(str(parameters["ssid"]))
+            self.key = quote(str(parameters["key"]))
+            self.ap_mac = quote(str(parameters["ap_mac"]))
+            self.country = quote(str(parameters["country"]).lower())
+            self.frequency = quote(str(parameters["frequency"]))
+            self.frequency_mcc = quote(str(parameters["frequency_mcc"]))
+            self.ip_address = quote(str(parameters["ip"]))
+            self.subnet = quote(str(parameters["subnet"]))
+            self.tx_power = quote(str(parameters["tx_power"]))
+            self.mode = quote(str(parameters["mode"]))
+            self.routing = quote(str(parameters["routing"]))
+            self.priority = quote(str(parameters["priority"]))
+            self.role = quote(str(parameters["role"]))
+            # not currently in json mesh_settings
+            self.mesh_vif = "wlp1s0"
```

Comment:
Later, May be we can get this added in drone manifest json

Reply:
We have NATS.io interface to support configuration changes.  If manifest.json will be used to hardcode settings for Mesh network --> drone side needs to send settings/apply to comms_controller to get new settings in use.